src/index.ts

```diff
@@ -1,45 +1,186 @@
+  [P in K[number]]: T[P];
+};
+
+type OmitResult<T extends Record<string, unknown>, K extends (keyof T)[]> = {
+  [P in Exclude<keyof T, K>]: T[P];
+};
+
+function _(selector: string) {
+  const element = document.querySelector<HTMLElement>(selector);
+  if (element === null) throw new Error("No Element");
+
+  const show = () => {
+    element.style.visibility = "visible";
+  };
+
+  const hide = () => {
+    element.style.visibility = "hidden";
+  };
+
+  const addEvent = <K extends keyof HTMLElementEventMap>(
+    type: K,
+    listener: (this: HTMLElement, event: HTMLElementEventMap[K]) => any
+  ) => {
+    element.addEventListener(type, listener);
+  };
+
+  element.show = show;
+  element.hide = hide;
+  element.addEvent = addEvent;
+
+  return element;
+}
+
+type Function<T = unknown, R = unknown> = (...args: T[]) => R;
+
+type FetchResource = string | URL | Request;
+
+type FetchOptions = {
+  method?: "GET" | "POST" | "PUT" | "DELETE";
+  headers?: Headers;
+  body?:
+    | string
+    | Blob
+    | ArrayBuffer
+    | DataView
+    | FormData
+    | URLSearchParams
+    | ReadableStream;
+};
+
 module _ {
-	export function fetch() {
-		return {};
-	}
+  export function fetch(
+    resource: FetchResource,
+    options?: FetchOptions
+  ): Promise<Response> {
+    return fetch(resource, options);
+  }
+
+  export function isNull(value: unknown): boolean {
+    return value === null;
+  }
+
+  export function isNil(value: unknown): boolean {
+    return value === null || value === undefined;
+  }
+
+  export function isNumber(value: unknown): boolean {
+    return typeof value === "number";
+  }
+
+  export function isFunction(value: unknown): boolean {
+    return typeof value === "function";
+  }
 
-	export function isNull() {}
+  export function shuffle<T>(array: T[]): T[] {
+    const length = array === null ? 0 : array.length;
+    if (!length) return [];
+    let index = -1;
+    const lastIndex = length - 1;
+    const result = [...array];
+    while (++index < length) {
+      const rand = index + Math.floor(Math.random() * (lastIndex - index + 1));
+      const value = result[rand];
+      result[rand] = result[index];
+      result[index] = value;
+    }
+    return result;
+  }
 
-	export function isNil() {}
+  export function pick<
+    T extends Record<string, unknown>,
+    K extends (keyof T)[]
+  >(object: T, paths: K): PickResult<T, K> {
+    const result = paths.reduce(
+      (prev, cur) => ({ ...prev, [cur]: object[cur] }),
+      {} as T
+    );
+    return result;
+  }
 
-	export function isNumber() {}
+  export function omit<
+    T extends Record<string, unknown>,
+    K extends (keyof T)[]
+  >(object: T, paths: K): OmitResult<T, K> {
+    const result = { ...object };
+    paths.forEach((key) => {
+      delete result[key];
+    });
+    return result;
+  }
 
-	export function isFunction() {}
+  export function memoize<T, R>(
+    func: Function<T, R>,
+    resolver?: Function<T, string>
```

Comment:
c: resolver의 value는 모든 원시 타입이 될 수 있을 것 같습니다.

```suggestion
    resolver?: Function<T, string | number | boolean | null | undefined | symbol | bigint>
```

Reply:
고마워요 👍👍👍 적용해봤습니다~ https://github.com/woowacourse/ts-module/pull/4/commits/19d056988011a29dec98571b8a660791a70e84f7